cmd/snapshot.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/0xPolygonHermez/zkevm-node"
+	"github.com/0xPolygonHermez/zkevm-node/config"
+	"github.com/0xPolygonHermez/zkevm-node/log"
+	pg "github.com/habx/pg-commands"
+	"github.com/urfave/cli/v2"
+)
+
+var snapshotFlags = []cli.Flag{
+	&configFileFlag,
+}
+
+func snapshot(ctx *cli.Context) error {
+	// Load config
+	c, err := config.Load(ctx, false)
+	if err != nil {
+		log.Error("error loading configuration. Error: ", err)
+		return err
+	}
+	setupLog(c.Log)
+
+	port, err := strconv.Atoi(c.StateDB.Port)
+	if err != nil {
+		log.Error("error converting port to int. Error: ", err)
+		return err
+	}
+	dump, err := pg.NewDump(&pg.Postgres{
+		Host:     c.StateDB.Host,
+		Port:     port,
+		DB:       c.StateDB.Name,
+		Username: c.StateDB.User,
+		Password: c.StateDB.Password,
+	})
+	if err != nil {
+		log.Error("error: ", err)
+		return err
+	}
+	log.Info("StateDB snapshot is being created...")
+	dump.SetFileName(fmt.Sprintf(`%v_%v_%v_%v.sql.tar.gz`, dump.DB, time.Now().Unix(), zkevm.Version, zkevm.GitRev))
```

Comment:
the result of executing a `pg_dump`is not a compressed tar file is:
 [Script dumps are plain-text files containing the SQL commands ](https://www.postgresql.org/docs/current/app-pgdump.html)

Reply:
As I can read this in the doc: 
`The most flexible output file formats are the “custom” format (-Fc) and the “directory” format (-Fd). They allow for selection and reordering of all archived items, support parallel restoration, and are compressed by default.`
By default the library uses custom format which should compress the dump. 
`pgDumpDefaultFormat = "c"`

Reply:
As far as I see it is compressed but not at the highest level.
`For the custom and directory archive formats, this specifies compression of individual table-data segments, and the default is to compress at a moderate level.`
I'm gonna try to force maximum compression

Reply:
@joanestebanr Now, it applies maximum compression (level 9). You can check it using command: `pg_restore -l <file>`

Reply:
Sorry, my fault, the format 'custom' have some plain text at begining! I checked your solution and with `-Z 9 `  reduce  80% size from the same DB with no compresion! (353Kb vs 81Kb)